File: Algorithms/C++/0055/0055.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> v(n, false);
        v[0] = true;
        int right = 0;
        for (int i = 0; i < n - 1; i++) {
            if (v[i]) {
                int t = i + nums[i];
                if (t >= n - 1) {
                    return true;
                } else if (t > right) {
                    for (int j = right + 1; j <= t; j++) {
                        v[j] = true;
                    }
                    right = t;
                }
            }
        }
        return v[n - 1];
    }
};

File: Algorithms/Python/0055/0055.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        flag = [False] * n
        flag[0] = True
        right = 0
        for i in range(n - 1):
            if i > right:
                break
            t = i + nums[i]
            if t >= n - 1:
                flag[-1] = True
                break
            elif t > right:
                for j in range(right + 1, t + 1):
                    flag[j] = True
                right = t
        return flag[-1]
